src/app/system/external-services/amazon-s3/edit-amazon-s3/edit-amazon-s3.component.ts

```diff
@@ -0,0 +1,71 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+/** Custom Services */
+import { SystemService } from 'app/system/system.service';
+
+@Component({
+  selector: 'mifosx-edit-amazon-s3',
+  templateUrl: './edit-amazon-s3.component.html',
+  styleUrls: ['./edit-amazon-s3.component.scss']
+})
+
+/**
+ * Edit Amazon S3 Component.
+ */
+export class EditAmazonS3Component implements OnInit {
+
+  /** Amazon S3 Configuration data */
+  amazonS3ConfigurationData: any;
+  /** Amazon S3 Configuration Form */
+  amazonS3ConfigurationForm: FormGroup;
+
+  /**
+   * Retrieves the Amazon S3 configuration data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {SystemService} systemService Accounting Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private systemService: SystemService,
+              private route: ActivatedRoute,
+              private router: Router) {
+    this.route.data.subscribe((data: { amazonS3Configuration: any }) => {
+      this.amazonS3ConfigurationData = data.amazonS3Configuration;
+    });
+  }
+
+  /**
+   * Creates the Amazon S3 configuration form.
+   */
+  ngOnInit() {
+    this.createEditAmazonS3ConfigurationForm();
+  }
+
+  /**
+   * Creates the edit Amazon S3 configuration form.
```

Comment:
```suggestion
   * Creates Amazon S3 configuration form.
```
No need to mention edit, make changes in all files necessary.